src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -140,7 +141,9 @@ public List<Car> getAllByDriver(Long driverId) {
             throw new DataProcessingException(
                     "Couldn't get a list of cars by driver with id " + driverId, e);
         }
-        cars.forEach(this::insertDriversToCar);
+        for (Car car: cars) {
+            insertDriversToCar(car);
+        }
```

Comment:
It was not fixed.

On the `get` call you are performing the insertion of the drivers to the **DB**.
Should it work like that?

Instead, you need to retrieve cars:
`getDriversByCar`

and then
`car.set(drivers)`

<img width="412" alt="image" src="https://github.com/mate-academy/jv-join/assets/68386737/8b376c85-8da1-4fea-a107-67b442b92145">

Reply:
On second thought, I realized I have no idea why I even used this method here, maybe, since I had big problems with this task I used different combinations and somehow ended with this.